test/transduce.js

```diff
@@ -40,6 +41,9 @@ describe('transduce', function() {
     eq(R.transduce(R.filter(isOdd), R.flip(R.append), [],  [1, 2, 3, 4]), [1, 3]);
     eq(R.transduce(R.compose(R.map(add(1)), R.take(2)), R.flip(R.append), [],  [1, 2, 3, 4]), [2, 3]);
     eq(R.transduce(R.compose(R.filter(isOdd), R.take(1)), R.flip(R.append), [],  [1, 2, 3, 4]), [1]);
+    eq(R.transduce(R.compose(R.take(6), R.uniq(), R.map(square)), R.flip(R.append), [],  [1, 1, 2, 2, 3, 3, 4, 4]), [1, 4, 9]);
+    eq(R.transduce(R.compose(R.map(negate), R.uniqBy(Math.abs), R.map(square)), R.flip(R.append), [],  [-1, -5, 2, 10, 1, 2]), [1, 25, 4, 100]);
+    eq(R.transduce(R.compose(R.take(6), R.uniqWith(R.eqBy(String)), R.map(square)), R.flip(R.append), [],  [1, '1', '2', 2, 3, '3', '4', 4]), [1, 4, 9]);
```

Comment:
I would like to see some tests with the `take` at the other end of the composition, since that is really the only reason I can even see to make `uniq*` into a transducer.

For instance if

```js
R.transduce(R.compose(R.uniq(), R.map(square), R.take(3)), R.flip(R.append), [],  [1, 1, 2, 2, 3, 3, 4, 4])
```

doesn't yield `[1, 4, 9]`, then we need to investigate carefully.

Reply:
Thanks for the quick feedback. I've made the requested change and amended the commit.

Reply:
> > I would like to see some tests with the take at the other end of the composition, since that is really the only reason I can even see to make uniq* into a transducer.

I can also think of a list of unique numbers in which case it would look something like:

```
compose(filter(is_even), uniq)
```

Suppose this would work as expected though.

Reply:
Added the suggested testcase.